### PR TITLE
ticdc: add binary_encoding_method to openapi v2

### PR DESCRIPTION
**Pull request by CharlesCheung96**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [ ] v7.4 (TiDB 7.4 versions)
- [ ] v7.3 (TiDB 7.3 versions)
- [ ] v7.2 (TiDB 7.2 versions)
- [x] v7.1 (TiDB 7.1 versions)
- [ ] v7.0 (TiDB 7.0 versions)
- [x] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-cn/pull/14877
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Review by nongfushanquan — APPROVED** [review]

**Review by sdojjy — APPROVED** [review]

File: ticdc/ticdc-changefeed-config.md [hunk 1]
```diff
@@ -224,4 +234,25 @@ sasl-oauth-scopes = ["producer.kafka", "consumer.kafka"]
...
+# The default value is 64MB.
+file-size = 67108864
+
+# The retention duration of the file, which takes effect only when date-separator is configured to day. Suppose that 'file-expiration-days = 1', TiCDC will clean up the files before 2023/12/01 (note: not including 2023/12/01) at 2023/12/02 00:00:00.
+# The default value is 0, which means file cleaning is disabled. 
+file-expiration-days = 0
+# This parameter controls the running period of the scheduled clearing task. It is compatible with the crontab configuration and the format is `Second | Minute | Hour | Dom | Month | DowOptional`.
+# The default value is "0 0 2 * * *".
+file-cleanup-cron-spec = "0 0 2 * * *"
+# The concurrency of uploading a single file.
```
Comment (qiancai) [inline ticdc/ticdc-changefeed-config.md:255]:
```suggestion
# The concurrency for uploading a single file.
```

File: ticdc/ticdc-changefeed-config.md [hunk 2]
```diff
@@ -224,4 +234,25 @@ sasl-oauth-scopes = ["producer.kafka", "consumer.kafka"]
...
+# The default value is 2s.
+flush-interval = "2s"
+# A data change file is stored to cloud storage if the number of bytes exceeds the value of this parameter.
+# The default value is 64MB.
+file-size = 67108864
+
+# The retention duration of the file, which takes effect only when date-separator is configured to day. Suppose that 'file-expiration-days = 1', TiCDC will clean up the files before 2023/12/01 (note: not including 2023/12/01) at 2023/12/02 00:00:00.
+# The default value is 0, which means file cleaning is disabled. 
+file-expiration-days = 0
+# This parameter controls the running period of the scheduled clearing task. It is compatible with the crontab configuration and the format is `Second | Minute | Hour | Dom | Month | DowOptional`.
```
Comment (qiancai) [inline ticdc/ticdc-changefeed-config.md:252]:
```suggestion
# The running cycle of the scheduled cleanup task, compatible with the crontab configuration, with a format of `Second | Minute | Hour | Dom | Month | DowOptional`
```

File: ticdc/ticdc-changefeed-config.md [hunk 3]
```diff
@@ -224,4 +234,25 @@ sasl-oauth-scopes = ["producer.kafka", "consumer.kafka"]
...
+# The default value is "0 0 2 * * *".
```
Comment (qiancai) [inline ticdc/ticdc-changefeed-config.md:253]:
```suggestion
# The default value is "0 0 2 * * *", which means that the cleanup task is executed every day at 2 AM.
```

File: ticdc/ticdc-open-api-v2.md [hunk 4]
```diff
@@ -288,6 +288,9 @@ The `consistent` parameters are described as follows:
 | `level`          | `STRING` type. The consistency level of the replicated data. (Optional)    |
 | `max_log_size`   | `UINT64` type. The maximum value of redo log. (Optional)      |
 | `storage`        | `STRING` type. The destination address of the storage. (Optional)            |
+| `encoding_worker_num` | `INT` type, The number of encoding workers in redo. (Optional)             |
+| `flush_worker_num`    | `INT` type, The number of flushing workers in redo. (Optional)             |
```
Comment (qiancai) [inline ticdc/ticdc-open-api-v2.md:292]:
```suggestion
| `use_file_backend` | `BOOL` type. Specifies whether to store the redo log in a local file. (Optional) |
| `encoding_worker_num` | `INT` type. The number of encoding workers in the redo module. (Optional)             |
| `flush_worker_num`    | `INT` type. The number of flushing workers in the redo module. (Optional)             |
```

File: ticdc/ticdc-open-api-v2.md [hunk 5]
```diff
@@ -344,10 +347,11 @@ The `sink.csv` parameters are described as follows:
 
 | Parameter name | Description |
```
Comment (qiancai) [inline ticdc/ticdc-open-api-v2.md:348]: 同中文比， L337 后面少了 `only_output_updated_columns` 和 `cloud_storage_config`

File: ticdc/ticdc-open-api-v2.md [hunk 6]
```diff
@@ -364,6 +368,17 @@ The `sink.csv` parameters are described as follows:
 | `partition` | `STRING` type. The target partition for dispatching events.    |
 | `topic`     | `STRING` type. The target topic for dispatching events.        |
 
+
+`sink.cloud_storage_config`  parameters are described as follows:
```
Comment (qiancai) [inline ticdc/ticdc-open-api-v2.md:372]:
```suggestion
`sink.cloud_storage_config`  parameters are described as follows:

```

File: ticdc/ticdc-open-api-v2.md [hunk 7]
```diff
@@ -364,6 +368,17 @@ The `sink.csv` parameters are described as follows:
...
+| Parameter name | Description |
+|:-----------------|:---------------------------------------|
+| `worker_count`   | `INT` type, Concurrency for saving data changes to cloud storage in the downstream.  |
+| `flush_interval`   | `STRING` type, Interval for saving data changes to cloud storage in the downstream. |
```
Comment (qiancai) [inline ticdc/ticdc-open-api-v2.md:376]:
```suggestion
| `worker_count`   | `INT` type. The concurrency for saving data changes to the downstream cloud storage.  |
| `flush_interval`   | `STRING` type. The interval for saving data changes to the downstream cloud storage. |
```

File: ticdc/ticdc-open-api-v2.md [hunk 8]
```diff
@@ -364,6 +368,17 @@ The `sink.csv` parameters are described as follows:
...
+| `file_size`   | `INT` type, a data change file is stored to cloud storage if the number of bytes exceeds the value of this parameter. |
+| `file_expiration_days`   | `INT` type, The retention duration of the file, which takes effect only when date-separator is configured to day. |
```
Comment (qiancai) [inline ticdc/ticdc-open-api-v2.md:378]:
```suggestion
| `file_size`   | `INT` type. A data change file is saved to the cloud storage when the number of bytes in this file exceeds the value of this parameter. |
| `file_expiration_days`   | `INT` type. The duration to retain files, which takes effect only when `date-separator` is configured as day. |
```

File: ticdc/ticdc-open-api-v2.md [hunk 9]
```diff
@@ -364,6 +368,17 @@ The `sink.csv` parameters are described as follows:
...
+| `file_cleanup_cron_spec`   | `STRING` type, This parameter controls the running period of the scheduled clearing task. It is compatible with the crontab configuration and the format is `Second | Minute | Hour | Dom | Month | DowOptional`.|
+| `flush_concurrency`   | `INT` type, The concurrency of uploading a single file.|
```
Comment (qiancai) [inline ticdc/ticdc-open-api-v2.md:380]:
```suggestion
| `flush_concurrency`   | `INT` type. The concurrency for uploading a single file. |
```

File: ticdc/ticdc-open-api-v2.md [hunk 10]
```diff
@@ -364,6 +368,17 @@ The `sink.csv` parameters are described as follows:
...
 
+
+`sink.cloud_storage_config`  parameters are described as follows:
+| Parameter name | Description |
+|:-----------------|:---------------------------------------|
+| `worker_count`   | `INT` type, Concurrency for saving data changes to cloud storage in the downstream.  |
+| `flush_interval`   | `STRING` type, Interval for saving data changes to cloud storage in the downstream. |
+| `file_size`   | `INT` type, a data change file is stored to cloud storage if the number of bytes exceeds the value of this parameter. |
+| `file_expiration_days`   | `INT` type, The retention duration of the file, which takes effect only when date-separator is configured to day. |
+| `file_cleanup_cron_spec`   | `STRING` type, This parameter controls the running period of the scheduled clearing task. It is compatible with the crontab configuration and the format is `Second | Minute | Hour | Dom | Month | DowOptional`.|
```
Comment (qiancai) [inline ticdc/ticdc-open-api-v2.md:379]:
```suggestion
| `file_cleanup_cron_spec`   | `STRING` type. The running cycle of the scheduled cleanup task, compatible with the crontab configuration, with a format of `Second | Minute | Hour | Dom | Month | DowOptional`. |
```

File: ticdc/ticdc-server-config.md [hunk 11]
```diff
@@ -39,6 +39,8 @@ data-dir = ""
 gc-ttl = 86400 # 24 h
 tz = "System"
 cluster-id = "default"
+# This variable specifies the maximum memory threshold for tuning GOGC: Setting a smaller threshold will increase the GC frequency. Setting a larger threshold will reduce GC frequency and allow TiCDC process to consume more memory resources. When the memory exceeds this threshold, GOGC Tuner stops working. The default value is 0, indicating that GOGC Tuner is disabled.
```
Comment (qiancai) [inline ticdc/ticdc-server-config.md:42]:
```suggestion
# This parameter specifies the maximum memory threshold (in bytes) for tuning GOGC: Setting a smaller threshold increases the GC frequency. Setting a larger threshold reduces GC frequency and consumes more memory resources for the TiCDC process. Once the memory usage exceeds this threshold, GOGC Tuner stops working. The default value is 0, indicating that GOGC Tuner is disabled.
```

File: ticdc/ticdc-changefeed-config.md [hunk 12]
```diff
@@ -224,4 +234,25 @@ sasl-oauth-scopes = ["producer.kafka", "consumer.kafka"]
...
+# The default value is 16.
+worker-count = 16
+# Interval for saving data changes to cloud storage in the downstream.
+# The default value is 2s.
+flush-interval = "2s"
+# A data change file is stored to cloud storage if the number of bytes exceeds the value of this parameter.
+# The default value is 64MB.
+file-size = 67108864
+
+# The retention duration of the file, which takes effect only when date-separator is configured to day. Suppose that 'file-expiration-days = 1', TiCDC will clean up the files before 2023/12/01 (note: not including 2023/12/01) at 2023/12/02 00:00:00.
```
Comment (qiancai) [inline ticdc/ticdc-changefeed-config.md:249]:
```suggestion
# The duration to retain files, which takes effect only when `date-separator` is configured as `day`. Assume that `file-expiration-days = 1` and `file-cleanup-cron-spec = "0 0 0 * * *"`, then TiCDC performs daily cleanup at 00:00:00 for files saved beyond 24 hours. For example, at 00:00:00 on 2023/12/02, TiCDC cleans up files generated before 2023/12/01, while files generated on 2023/12/01 remain unaffected.
```

File: ticdc/ticdc-changefeed-config.md [hunk 13]
```diff
@@ -224,4 +233,25 @@ sasl-oauth-scopes = ["producer.kafka", "consumer.kafka"]
...
+worker-count = 16
+# The interval for saving data changes to the downstream cloud storage.
+# The default value is "2s".
+flush-interval = "2s"
+# A data change file is saved to the cloud storage when the number of bytes in this file exceeds `file-size`.
+# The default value is 67108864 (this is, 64 MiB).
+file-size = 67108864
+
+# The duration to retain files, which takes effect only when `date-separator` is configured as `day`. Assume that `file-expiration-days = 1` and `file-cleanup-cron-spec = "0 0 0 * * *"`, then TiCDC performs daily cleanup at 00:00:00 for files saved beyond 24 hours. For example, at 00:00:00 on 2023/12/02, TiCDC cleans up files generated before 2023/12/01, while files generated on 2023/12/01 remain unaffected.
+# The default value is 0, which means file cleaning is disabled.
```
Comment (qiancai) [inline ticdc/ticdc-changefeed-config.md:249]:
```suggestion
# The default value is 0, which means file cleanup is disabled. 
```

File: ticdc/ticdc-changefeed-config.md [hunk 14]
```diff
@@ -224,4 +233,25 @@ sasl-oauth-scopes = ["producer.kafka", "consumer.kafka"]
...
+file-size = 67108864
+
+# The duration to retain files, which takes effect only when `date-separator` is configured as `day`. Assume that `file-expiration-days = 1` and `file-cleanup-cron-spec = "0 0 0 * * *"`, then TiCDC performs daily cleanup at 00:00:00 for files saved beyond 24 hours. For example, at 00:00:00 on 2023/12/02, TiCDC cleans up files generated before 2023/12/01, while files generated on 2023/12/01 remain unaffected.
+# The default value is 0, which means file cleanup is disabled. 
+file-expiration-days = 0
+# The running cycle of the scheduled cleanup task, compatible with the crontab configuration, with a format of `Second | Minute | Hour | Dom | Month | DowOptional`
+# The default value is "0 0 2 * * *", which means that the cleanup task is executed every day at 2 AM.
+file-cleanup-cron-spec = "0 0 2 * * *"
+# The concurrency for uploading a single file.
+# The default value is 1.
```
Comment (ran-huang) [inline ticdc/ticdc-changefeed-config.md:255]:
```suggestion
# The default value is 1, which means concurrency is disabled.
```

File: ticdc/ticdc-open-api-v2.md [hunk 15]
```diff
@@ -288,6 +288,10 @@ The `consistent` parameters are described as follows:
 | `level`          | `STRING` type. The consistency level of the replicated data. (Optional)    |
 | `max_log_size`   | `UINT64` type. The maximum value of redo log. (Optional)      |
 | `storage`        | `STRING` type. The destination address of the storage. (Optional)            |
+| `use_file_backend` | `BOOL` type. Specifies whether to store the redo log in a local file. (Optional) |
+| `encoding_worker_num` | `INT` type. The number of encoding workers in the redo module. (Optional)             |
```
Comment (ran-huang) [inline ticdc/ticdc-open-api-v2.md:292]:
```suggestion
| `encoding_worker_num` | `INT` type. The number of encoding and decoding workers in the redo module. (Optional)             |
```

File: ticdc/ticdc-open-api-v2.md [hunk 16]
```diff
@@ -344,10 +350,11 @@ The `sink.csv` parameters are described as follows:
...
-| `include_commit_ts` | `BOOLEAN` type. Whether to include commit-ts in CSV rows. The default value is `false`. |
-| `null`              | `STRING` type. The character that is displayed when a CSV column is null. The default value is `\N`. |
-| `quote`             | `STRING` type. The quotation character used to surround fields in the CSV file. If the value is empty, no quotation is used. The default value is `"`. |
+| `delimiter`              | `STRING` type. The character used to separate fields in the CSV file. The value must be an ASCII character and defaults to `,`.     |
+| `include_commit_ts`      | `BOOLEAN` type. Whether to include commit-ts in CSV rows. The default value is `false`. |
+| `null`                   | `STRING` type. The character that is displayed when a CSV column is null. The default value is `\N`. |
+| `quote`                  | `STRING` type. The quotation character used to surround fields in the CSV file. If the value is empty, no quotation is used. The default value is `"`. |
+| `binary_encoding_method` | `STRING` type. The encoding method of binary data, which can be 'base64' or 'hex'. The default value is 'base64'. |
```
Comment (ran-huang) [inline ticdc/ticdc-open-api-v2.md:357]:
```suggestion
| `binary_encoding_method` | `STRING` type. The encoding method of binary data, which can be `"base64"` or `"hex"`. The default value is `"base64"`. |
```

File: ticdc/ticdc-open-api-v2.md [hunk 17]
```diff
@@ -364,6 +371,18 @@ The `sink.csv` parameters are described as follows:
 | `partition` | `STRING` type. The target partition for dispatching events.    |
 | `topic`     | `STRING` type. The target topic for dispatching events.        |
 
+
```
Comment (ran-huang) [inline ticdc/ticdc-open-api-v2.md:374]:
```suggestion
```

File: ticdc/ticdc-open-api-v2.md [hunk 18]
```diff
@@ -364,6 +371,18 @@ The `sink.csv` parameters are described as follows:
...
+`sink.cloud_storage_config`  parameters are described as follows:
+
+| Parameter name | Description |
+|:-----------------|:---------------------------------------|
+| `worker_count`   | `INT` type. The concurrency for saving data changes to the downstream cloud storage.  |
+| `flush_interval`   | `STRING` type. The interval for saving data changes to the downstream cloud storage. |
+| `file_size`   | `INT` type. A data change file is saved to the cloud storage when the number of bytes in this file exceeds the value of this parameter. |
+| `file_expiration_days`   | `INT` type. The duration to retain files, which takes effect only when `date-separator` is configured as `day`. |
+| `file_cleanup_cron_spec`   | `STRING` type. The running cycle of the scheduled cleanup task, compatible with the crontab configuration, with a format of `Second | Minute | Hour | Dom | Month | DowOptional`. |
```
Comment (ran-huang) [inline ticdc/ticdc-open-api-v2.md:383]:
```suggestion
| `file_cleanup_cron_spec`   | `STRING` type. The running cycle of the scheduled cleanup task, compatible with the crontab configuration, with a format of `Second \| Minute \| Hour \| Dom \| Month \| DowOptional`. |
```

File: ticdc/ticdc-changefeed-config.md [hunk 19]
```diff
@@ -224,4 +233,25 @@ sasl-oauth-scopes = ["producer.kafka", "consumer.kafka"]
...
+# The default value is "2s".
+flush-interval = "2s"
+# A data change file is saved to the cloud storage when the number of bytes in this file exceeds `file-size`.
+# The default value is 67108864 (this is, 64 MiB).
+file-size = 67108864
+
+# The duration to retain files, which takes effect only when `date-separator` is configured as `day`. Assume that `file-expiration-days = 1` and `file-cleanup-cron-spec = "0 0 0 * * *"`, then TiCDC performs daily cleanup at 00:00:00 for files saved beyond 24 hours. For example, at 00:00:00 on 2023/12/02, TiCDC cleans up files generated before 2023/12/01, while files generated on 2023/12/01 remain unaffected.
+# The default value is 0, which means file cleanup is disabled. 
+file-expiration-days = 0
+# The running cycle of the scheduled cleanup task, compatible with the crontab configuration, with a format of `Second | Minute | Hour | Dom | Month | DowOptional`
```
Comment (qiancai) [inline ticdc/ticdc-changefeed-config.md:251]:
```suggestion
# The running cycle of the scheduled cleanup task, compatible with the crontab configuration, with a format of `<Second> <Minute> <Hour> <Day of the month> <Month> <Day of the week (Optional)>`
```

**Review by ran-huang — APPROVED** [review]

**Review by qiancai — APPROVED** [review]

File: ticdc/ticdc-open-api-v2.md [hunk 20]
```diff
@@ -333,6 +337,8 @@ The `sink` parameters are described as follows:
 | `transaction_atomicity` | `STRING` type. The atomicity level of the transaction. (Optional)                                                                                                                                              |
 | `only_output_updated_columns` | `BOOLEAN` type. For MQ sinks using the `canal-json` or `open-protocol` protocol, you can specify whether only output the modified columns. The default value is `false`. (Optional) |
```
Comment (CharlesCheung96) [inline ticdc/ticdc-open-api-v2.md:339]:
```suggestion
```

File: ticdc/ticdc-changefeed-config.md [hunk 21]
```diff
@@ -208,9 +208,18 @@ flush-interval = 2000
 # The storage URI of the redo log.
 # The default value is empty.
 storage = ""
-# Specifies whether to store the redo log in a file.
+# Specifies whether to store the redo log in a local file.
 # The default value is false.
 use-file-backend = false
+# The number of encoding workers in the redo module.
```
Comment (qiancai) [inline ticdc/ticdc-changefeed-config.md:214]:
```suggestion
# The number of encoding and decoding workers in the redo module.
```

File: ticdc/ticdc-open-api-v2.md [hunk 22]
```diff
@@ -287,6 +287,10 @@ The `consistent` parameters are described as follows:
 | `level`          | `STRING` type. The consistency level of the replicated data. (Optional)    |
 | `max_log_size`   | `UINT64` type. The maximum value of redo log. (Optional)      |
 | `storage`        | `STRING` type. The destination address of the storage. (Optional)            |
+| `use_file_backend` | `BOOL` type. Specifies whether to store the redo log in a local file. (Optional) |
+| `encoding_worker_num` | `INT` type. The number of encoding and decoding workers in the redo module. (Optional)             |
+| `flush_worker_num`    | `INT` type. The number of flushing workers in the redo module. (Optional)             |
+
```
Comment (qiancai) [inline ticdc/ticdc-open-api-v2.md:293]:
```suggestion
```

**Pull request merged**